test/extended/router/config_manager.go

```diff
@@ -52,8 +57,439 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:route.openshift.io][a
+					},
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationPassthrough,
+					},
+					Host: "passthrough.hapcm.test",
+					To: routev1.RouteTargetReference{
+						Name: "secure-service",
+						Kind: "Service",
+					},
+				},
+			},
+		}
+
+		for _, route := range routes {
+			_, err := oc.RouteClient().RouteV1().Routes(ns).Create(context.Background(), &route, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+
+		g.By("creating route Pods")
+		routerPods := []corev1.Pod{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "router-haproxy-cfgmgr",
+					Labels: map[string]string{
+						"test": "router-haproxy-cfgmgr",
+					},
+				},
+				Spec: corev1.PodSpec{
+					TerminationGracePeriodSeconds: utilpointer.Int64(1),
+					Containers: []corev1.Container{
+						{
+							Name:            "router",
+							Image:           routerImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Env: []corev1.EnvVar{
+								{
+									Name: "POD_NAMESPACE",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "metadata.namespace",
+										},
+									},
+								},
+							},
+							Args: []string{
+								"--namespace=$(POD_NAMESPACE)",
+								"-v=4",
+								"--haproxy-config-manager=true",
+								"--blueprint-route-labels=select=hapcm-blueprint",
+								"--labels=select=haproxy-cfgmgr",
+								"--stats-password=password",
+								"--stats-port=1936",
+								"--stats-user=admin",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+								{
+									ContainerPort: 443,
+								},
+								{
+									ContainerPort: 1936,
+									Name:          "stats",
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+						},
+					},
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "insecure-endpoint",
+					Labels: map[string]string{
+						"test":      "haproxy-cfgmgr",
+						"endpoints": "insecure-endpoint",
+					},
+				},
+				Spec: corev1.PodSpec{
+					TerminationGracePeriodSeconds: utilpointer.Int64(1),
+					Containers: []corev1.Container{
+						{
+							Name:  "test",
+							Image: image.LocationFor("registry.k8s.io/e2e-test-images/agnhost:2.40"),
+							Args:  []string{"netexec"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8080,
+									Name:          "http",
+								},
+								{
+									ContainerPort: 100,
+									Protocol:      corev1.ProtocolUDP,
+								},
+							},
+						},
+					},
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "secure-endpoint",
+					Labels: map[string]string{
+						"app": "secure-endpoint",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "server",
+							Image:   "k8s.gcr.io/e2e-test-images/nginx:1.15-2",
```

Comment:
Any reason you use k8s.gcr.io here and registry.k8s.io above, or why you use `image.LocationFor` above and not here?

Reply:
The validation test failed and complained about the agnhost not existing in the quay.io mirror. Using `LocationFor` for the nginx image now as well.